File: String/1759.Count Number of Homogenous Substrings/Solution.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        ans = k = 1
        for i in range(1, len(s)):
            if s[i - 1] == s[i]:
                k += 1
            else:
                k = 1
            ans += k
        return ans % int(1e9 + 7)

File: String/1759.Count Number of Homogenous Substrings/Solution.java
class Solution {
    public int countHomogenous(String s) {
        int ans = 1, k = 1, mod = (int)1e9 + 7;
        char[] cs = s.toCharArray();
        for (int i = 1; i < cs.length; i++) {
            if (cs[i - 1] == cs[i]) {
                k++;
            } else {
                k = 1;
            }
            ans = (ans + k) % mod;
        }
        return ans % mod;
    }
}